app/code/core/Mage/Customer/controllers/AccountController.php

```diff
@@ -655,7 +655,7 @@ public function confirmAction()
      */
     public function confirmationAction()
     {
-        $customer = $this->_getModel('customer/customer');
```

Comment:
is the _getModel method still used/necessary at this point?

Reply:
It's still used. Is it better to replace them all with `Mage::getModel()` and marked `_getModel()` as deprecated?

Reply:
If the _getModel is still necessary, why changing it to Mage::getModel()? just curiousity

Reply:
PHPStan thinks `$customer` is `Mage_Core_Model_Abstract` and complains 

> message: "#^Cannot call method sendNewAccountEmail\\(\\) on Mage_Core_Model_Abstract\\|false\\.$#"

Reply:
`instanceof Mage_Customer_Model_Customer` check could solve it. (?)

Reply:
I didn't know that. But for [green coding](https://www.ibm.com/blog/green-coding/) sake,  why waste another CPU cycle.

Reply:
Mhh, or simply ...

```
/** @var Mage_Customer_Model_Customer $customer **/
$customer = $this->_getModel(...);
$collection = $customer->
```

Reply:
@sreichel I am lost, care to explain simply or make a code suggestion so that I can hopefully see the point?

Reply:
Sorry, my bad. Provide code suggestion later.

Reply:
@kiatng as was at the wrong place in code ...

I thought about "green coding"  ...

- adding instanceof-check is only one more function call and takes only a few bytes in memory usage (tested with xhprof)
- adding a new temporally variable may take more memory (not tested yet)
- adding inline DocBlock works, but you tell phpstan how to behave ... in my forked test-project i tried to avoid it)

Solution 1: add inline Docs

```
        /** @var Mage_Customer_Model_Customer $customer */ # added line
        $customer = $this->_getModel('customer/customer');
```

This works, but is just workaround b/c getModel(...) could return "false"  too. (it does not, but from static analyzis it could)

Solution 2: instanceof

```
        if (!$customer instanceof Mage_Customer_Model_Customer) {
            throw new Exception('Customer model must to extend Mage_Customer_Model_Customer');
        }
```

I'd prefer 2nd one.

Reply:
In the controller's context, if `Mage::getModel('customer/customer')` return `false`, something is seriously off, an exception should be thrown. So I think the fix I have now is the simplest.